[MNT] Deprecate LogCpTransformer in favour of LogTransformer with C parameter (#970)
feature-engine/feature_engine: docs/user_guide/transformation/LogTransformer.rst

.. note::

>     By default (``C=0``), the logarithm can only be applied to positive values. If the variable contains 0 or negative values, the transformer will return an error, unless you use the ``C`` parameter described below.

Reviewer: ```suggestion
    The logarithm can only be applied to positive values; if the variable contains 0 or negative values, the transformer will raise an error. To avoid this, add a constant to the variables by using the `C` parameter (more details below).
```

> To transform non-positive variables, pass a value to the ``C`` parameter to shift

Reviewer: ```suggestion
```

> the data points towards positive values, as shown in the section below.

Reviewer: ```suggestion
```

Reviewer: We can remove these sentences, since it's included in the note above.

    Following the transformations with scatter plots and residual analysis of the regression models helps understand if the transformations are useful in our regression analysis.

Transforming non-positive variables with C

> ------------------------------------------

Reviewer: We need to replace the lines under the hading -- with sublevel heading so ˜˜˜, so that this heading appears under the heading Python implementaiton

:class:`LogTransformer()` can also transform variables that contain zero or
negative values, by adding a constant ``C`` before applying the logarithm,
i.e. ``log(x + C)``.

Let's load the diabetes dataset that comes with scikit-learn, which contains
variables with negative values:

.. code:: python

    from sklearn.model_selection import train_test_split

>     from sklearn.datasets import load_diabetes

Reviewer: this is a nitpick, but since we are editing still, it's nicer to have imports in alphabetical order, so datasets, before model_selection
